### ISSUE TITLE
UnsupportedOperation when running in Jupyter Notebook

### ISSUE THREAD

**Issue by pwnorbitals** [Gallopsled/pwntools#1778]
In Kali Linux, Jupyter Notebook running as root (yeah I know, but that's a bug for another time)
Happens consistently at import
Same issue in JupyterLab (not as root this time :) )

<details>
<summary>Traceback</summary>

```
---------------------------------------------------------------------------
UnsupportedOperation                      Traceback (most recent call last)
<ipython-input-1-6d46e8d02821> in <module>
      2 import threading
      3 import subprocess
----> 4 from pwn import *

/usr/local/lib/python3.9/dist-packages/pwn/__init__.py in <module>
      2 from __future__ import absolute_import
      3 
----> 4 from pwn.toplevel import *
      5 
      6 pwnlib.args.initialize()

/usr/local/lib/python3.9/dist-packages/pwn/toplevel.py in <module>
     18 from pprint import pprint
     19 
---> 20 import pwnlib
     21 from pwnlib import *
     22 from pwnlib.asm import *

/usr/local/lib/python3.9/dist-packages/pwnlib/__init__.py in <module>
     41 
     42 for module in __all__:
---> 43     importlib.import_module('.%s' % module, 'pwnlib')

/usr/lib/python3.9/importlib/__init__.py in import_module(name, package)
    125                 break
    126             level += 1
--> 127     return _bootstrap._gcd_import(name[level:], package, level)
    128 
    129 

/usr/local/lib/python3.9/dist-packages/pwnlib/args.py in <module>
     60 import sys
     61 
---> 62 from pwnlib import term
     63 from pwnlib.context import context
     64 

/usr/local/lib/python3.9/dist-packages/pwnlib/term/__init__.py in <module>
      4 import sys
      5 
----> 6 from pwnlib.term import completer
      7 from pwnlib.term import key
      8 from pwnlib.term import keymap

/usr/local/lib/python3.9/dist-packages/pwnlib/term/completer.py in <module>
      5 import re
      6 
----> 7 from pwnlib.term import readline
      8 
      9 

/usr/local/lib/python3.9/dist-packages/pwnlib/term/readline.py in <module>
     11 from pwnlib.term import keymap as km
     12 from pwnlib.term import term
---> 13 from pwnlib.term import text
     14 
     15 cursor = text.reverse

/usr/local/lib/python3.9/dist-packages/pwnlib/term/text.py in <module>
    124 
    125 tether = sys.modules[__name__]
--> 126 sys.modules[__name__] = Module()

/usr/local/lib/python3.9/dist-packages/pwnlib/term/text.py in __init__(self)
     41             'white': 7,
     42             }
---> 43         self._reset = termcap.get('reset')
     44         self._attributes = {}
     45         for x, y in [('italic'   , 'sitm'),

/usr/local/lib/python3.9/dist-packages/pwnlib/term/unix_termcap.py in get(cap, *args, **kwargs)
     23 
     24     if cache is None:
---> 25         init()
     26     s = cache.get(cap)
     27     if not s:

/usr/local/lib/python3.9/dist-packages/pwnlib/term/unix_termcap.py in init()
     49         # Fix for BPython
     50         try:
---> 51             curses.setupterm()
     52         except curses.error as e:
     53             import traceback

UnsupportedOperation: fileno
```

</details>

**Comment by Arusekk** [comment]
Known bug, dup of #1671, see #826 #1339 #1632 for context; possibly resolved in newest version.
TL;DR: set `PWNLIB_NOTERM=1` in the environment to disable fancy terminal functionality altogether.

**Comment by pwnorbitals** [comment]
thanks for your quick response, sorry for the dup

**Comment by zachriggle** [comment]
@pwnorbitals how did you make that neat folding traceback?

@Arusekk looks like we just need to handle `UnsupportedOperation` in addition to `curses.error` for that setup code in order to (verbosely) work around the issue and at least let Pwntools to actually work.

https://github.com/Gallopsled/pwntools/blob/2086a3195425201619c4d33bb3c3ba8b53d1aa0a/pwnlib/term/unix_termcap.py#L58-L64

**Comment by pwnorbitals** [comment]
@zachriggle code is : 
< details >
< summary >Traceback</ summary >
` ` `
code
` ` `
</ details >
